File: Sorting/Sort Array By Parity/Approach-1-ExtraArray.py
class Solution(object):
    def sortArrayByParity(self, nums):
        result=[0]*len(nums)
        start=0        
        end=len(nums)-1

        for i in nums:
            if i%2==0:
                result[start]=i
                start+=1
            else :
                result[end]=i
                end-=1    

        return result   

File: Sorting/Sort Array By Parity/Approach-1-ExtraArray.java
class Solution {
    public int[] sortArrayByParity(int[] nums) {
     
        int n=nums.length;
        
        int[] result=new int[n];
        
        int start =0,end=n-1;
        for(int i=0;i<n;i++){
            if(nums[i]%2==0){
                result[start++]=nums[i];
            }
            else{
                result[end--]=nums[i];
            }
        }
        
        return result;
    }
}
